### ISSUE TITLE
Certain URLs wont load. Error is null. 

### ISSUE THREAD

**Issue by ardevd** [bumptech/glide#1675]
I'm having the exact same issue as #741. Exception is null and I have no idea why some of my images wont load. The URLs passed into Glide seem fine. 

```java
       @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            final Location location = mLocations.get(position);

            if (location.hasImage()) {
                String imageUrl = NetworkHelper.getImageUrl(getContext(),location.getImageName());
                holder.image.setVisibility(View.VISIBLE);
                // Use Glide for image loading
                Glide.with(getContext())
                        .load(imageUrl)
                        .diskCacheStrategy(DiskCacheStrategy.ALL)
                        .centerCrop()
                        .into(new GlideDrawableImageViewTarget(holder.image) {

                            @Override
                            public void onLoadFailed(Exception e, Drawable errorDrawable) {
                                super.onLoadFailed(e, errorDrawable);
                                e.printStackTrace();
                            }

                            @Override
                            public void onResourceReady(GlideDrawable resource, GlideAnimation<? super GlideDrawable> animation) {
                                super.onResourceReady(resource, animation);
                            }
                        });
            } else {
                // Due to the nature of the RecyclerView, we need to ensure we clear
                // out images to odd behavior.
                Glide.clear(holder.image);
                holder.image.setVisibility(View.GONE);
            }
        }
```

I've tried hardcoding one of the URLs that fails and it fails consistently so there must be something about those specific images Glide doesnt like though I cant quite see why. I can download them manually just fine.

Thanks!

**Comment by TWiStErRob** [comment]
I can see similar things here as well... :) I'm assuming you added the custom target just for logging. Please try the other methods on the [wiki](https://github.com/bumptech/glide/wiki/Debugging-and-Error-Handling), if all else fails try to clear the cache.

Can you share a public failing url we could investigate?

**Comment by archpoint** [comment]
Unfortunately, the URL in question is only available internally so that's not possible. However, I would be happy to provide the actual image file if that helps? 

I'll check the wiki and see if any of the troubleshooting efforts yield any results.

**Comment by TWiStErRob** [comment]
I don't think the actual bytes could be wrong, though looking for `skia` errors in LogCat may be a good first step. I was asking to see the headers and stuff. If it's private is it possible that it's IP/user-agent blocked or something similar? Enabling logging should help to see where Glide has a hiccup.

**Comment by archpoint** [comment]
The IP/User-agent is not blocked. All the requested images are from the same server so there should be no server-side issue blocking the image download. I'm seeing the issue regardless of whether I run it with an emulator or on an actual device.

**Comment by archpoint** [comment]
```
01-10 10:26:20.814  2933  3114 D skia    : libjpeg error 31 <Bogus DQT index 11> from output_message
01-10 10:26:20.814  2933  3114 D skia    : libjpeg error 31 <Bogus DQT index 11> from setjmp
01-10 10:26:20.814  2933  3114 D skia    : --- SkAndroidCodec::NewFromStream returned null
01-10 10:26:20.814  2933  3114 D skia    : libjpeg error 31 <Bogus DQT index 11> from output_message
01-10 10:26:20.814  2933  3114 D skia    : libjpeg error 31 <Bogus DQT index 11> from setjmp
01-10 10:26:20.814  2933  3114 D skia    : --- SkAndroidCodec::NewFromStream returned null
```

This is the relevant logcat output, so there is indeed some error output. I don't quite understand the error though. The jpeg images I'm trying to load where captured on an iPhone 7 and resized using Adobe Photoshop CC.

**Comment by TWiStErRob** [comment]
That sounds like either a JPEG format issue, or a bug in skia/libjpeg. We can't do anything about this on Glide level, unless you want to replace your decoder with a native one that supports your "Bogus" files.

**Comment by archpoint** [comment]
That's really weird. Maybe the web service back end manipulates the files someway, but it doesn't seem like it. Would it help if I attached one of the photos here? 

While it may not be a Glide issue, it would be nice if the Glide exception was something other than Null though.

**Comment by archpoint** [comment]
I used jpeginfo on the jpeg files and that returns an error as well, so it's certainly a jpeg format issue:

`Bogus DQT index 11  [ERROR] `

**Comment by TWiStErRob** [comment]
I think if Glide fetcher/decoder (i.e. Network stream/BitmapFactory) returns null then the error propagated to us is null, because there was no exception. I agree probably an `IllegalStateException("Decoder returned null")` would be more appropriate. @sjudd?

**Comment by sjudd** [comment]
Definitely a good point, we could pass back a custom error message. I agree "null" is confusing.

**Comment by XiaYL** [comment]
I also got this issue.It's strange that the image is normal with chrome,but load error with phone.here is my log bellow,picture url is [http://news.xinhuanet.com/politics/2016-12/09/129398225_14812941507631n.jpg](url)
![4zaj88d iaiubthe pvt0 0](https://cloud.githubusercontent.com/assets/15325562/22091533/3f5b13e0-de32-11e6-87ea-f52636d44201.png)

**Comment by TWiStErRob** [comment]
@XiaYL, no, you didn't get this issue; you clearly have an exception and not `null`. If you search for that, you'll end up on #432.

Your host is slow; if you're displaying lot of images from that host, increase the timeout.

**Comment by XiaYL** [comment]
@TWiStErRob, I just test the image url later and found the method onLoadException returned null, so I debuged the code, it finally ran into BitmapFactory , and reminded the inputstream was null, but i just got the stream indeed, then returned null, it's wired

**Comment by evgenii-paringer** [comment]
files with extension '.jpg' are loading, while files with extension '.jpeg' are failed with NullPointerException, I used, ImageViewTarget<Drawable>, please fix, because, I cannot rename uploaded avatar images, easily
